### PR TITLE
Another system config: allow_user_to_change_email_address

### PR DESCRIPTION
**Pull request by theronakpatel**
* Resolves: # https://github.com/nextcloud/server/issues/16823

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/provisioning_api/tests/Controller/UsersControllerTest.php [hunk 1]
```diff
@@ -4171,6 +4183,12 @@ public function testGetEditableFields(bool $allowedToChangeDisplayName, string $
 				$this->equalTo('allow_user_to_change_display_name'),
 				$this->anything()
 			)->willReturn($allowedToChangeDisplayName);
+		$this->config
+			->method('getSystemValue')
+			->with(
+				$this->equalTo('allow_user_to_change_email_address'),
+				$this->anything()
+			)->willReturn($allowedToChangeDisplayName);
```
Comment (akhil1508) [inline apps/provisioning_api/tests/Controller/UsersControllerTest.php:4191]: This has to be a different variable

File: lib/private/User/Backend.php [hunk 2]
```diff
@@ -47,14 +47,15 @@ abstract class Backend implements UserInterface {
...
-
+	public const SET_EMAILADDRESS = 4294967296;		// 1 << 32
 	protected $possibleActions = [
 		self::CREATE_USER => 'createUser',
 		self::SET_PASSWORD => 'setPassword',
 		self::CHECK_PASSWORD => 'checkPassword',
 		self::GET_HOME => 'getHome',
 		self::GET_DISPLAYNAME => 'getDisplayName',
 		self::SET_DISPLAYNAME => 'setDisplayName',
+		self::SET_EMAILADDRESS => 'setEMailAddress',
```
Comment (akhil1508) [inline lib/private/User/Backend.php:58]: @theronakpatel Where is this change coming from?

File: lib/public/IUser.php [hunk 3]
```diff
@@ -143,6 +143,14 @@ public function canChangePassword();
 	 */
 	public function canChangeDisplayName();
 
+	/**
+	 * check if the backend supports changing email addresses
+	 *
+	 * @return bool
+	 * @since 8.0.0
```
Comment (akhil1508) [inline lib/public/IUser.php:150]: Its since 28.0.0? It's a change we are introducing now..

File: tests/lib/User/UserTest.php [hunk 4]
```diff
@@ -373,6 +373,31 @@ public function testCanChangeDisplayName() {
...
+	public function testCanChangeEmailAddress() {
+		/**
+		 * @var Backend | MockObject $backend
+		 */
+		$backend = $this->createMock(\Test\Util\User\Dummy::class);
+
+		$backend->expects($this->any())
+			->method('implementsActions')
+			->willReturnCallback(function ($actions) {
+				if ($actions === \OC\User\Backend::SET_DISPLAYNAME) {
```
Comment (akhil1508) [inline tests/lib/User/UserTest.php:385]: SET_EMAILADDRESS?

File: lib/private/User/Backend.php (hunk 2 above)
Comment (theronakpatel) [reply lib/private/User/Backend.php:58]: @akhil1508 In this file actions are defined. We already have `setEMailAddress` function.

File: apps/provisioning_api/tests/Controller/UsersControllerTest.php (hunk 1 above)
Comment (theronakpatel) [reply apps/provisioning_api/tests/Controller/UsersControllerTest.php:4191]: Done.

File: lib/public/IUser.php (hunk 3 above)
Comment (theronakpatel) [reply lib/public/IUser.php:150]: IG, current is 27.0.0, so yes from 28.0.0 @akhil1508

File: tests/lib/User/UserTest.php (hunk 4 above)
Comment (theronakpatel) [reply tests/lib/User/UserTest.php:385]: yes @akhil1508

**Comment by theronakpatel** [comment]
@akhil1508 I updated https://github.com/nextcloud/server/pull/40796

**Pull request closed without merging**